### PR TITLE
Add find_or_create_not_completed_answer_survey

### LINKED ISSUE #491: Do not try to create answers_survey for survey if incomplete answers_survey already exists
**Current behavior**
- visit https://espoo.herokuapp.com/surveys
- click `answer button` for survey X with zero questions => redirected to show question page
- visit https://espoo.herokuapp.com/surveys
- click `answer button` for survey X with zero questions => Receives error message like the one bellow
image

**Expected behavior** 
- visit https://espoo.herokuapp.com/surveys
- click `answer button` for survey X with zero questions => redirected to show question page
- visit https://espoo.herokuapp.com/surveys
- click `answer button` for survey X with zero questions => edirected to show question page

### PR DESCRIPTION
**Pull request by PolinaStamenova**
fix #491

# Fix raising error "Should exist only one not completed by survey and user." 

- Create `find_or_create_not_completed_answer_survey`  that check if there is an existing `AnswersSurvey` (_not copleted_) and return it. In case it does NOT exist it will create a new one.

#### Only select the appropriate.

- [ ] **Model testing.**
- [ ] **Request testing and swagger doc update.**
- [ ] **System testing.**
- [ ] **No tests required.**

### CODE REVIEW

**Review by edimossilva — CHANGES_REQUESTED** [review]
Please add system tests for it

**Review by edimossilva — CHANGES_REQUESTED** [review]
@PolinaStamenova , my bad, I didn't notice that this won't work.
The first time the user is answering a survey it will create the first AnswersSurvey.
The second time the user is answering a survey it will create a second AnswersSurvey.

**Comment by edimossilva** [comment]
> @PolinaStamenova , my bad, I didn't notice that this won't work. The first time the user is answering a survey it will create the first AnswersSurvey. The second time the user is answering a survey it will create a second AnswersSurvey.

@PolinaStamenova , my bad I didn't pay attention to `if answer_survey&.not_completed?`. Please, just add specs for it

**Comment by PolinaStamenova** [comment]
> Please add system tests for it

I checked yesterday, but all test are passing. I am not sure what else can be tested?

File: backend/spec/requests/answers_survey_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  let(:user_student) { create(:user_student) }
+  let(:survey) { create(:survey_with_2_questions) }
+  let(:answers_survey_params) { { survey_id: survey.id } }
+
+  describe '#create' do
+    subject(:call) { post answers_surveys_path, params: answers_survey_params, headers: auth_headers(user: user_student) }
+
+    it 'response status' do
+      call
+      expect(response).to have_http_status :found
```
Comment (edimossilva) [inline backend/spec/requests/answers_survey_request_spec.rb:13]: :found status is returned, but in my mind should be :created, does it make sense?

I am not familiar with this kind of test so I have no strong opinion about this

**Review by edimossilva — APPROVED** [review]
Nice work, 
also good job refactoring other specs

**Pull request merged**